Tests/Marketplace/search_and_install_packs.py

```diff
@@ -33,233 +35,215 @@
+GITLAB_PACK_METADATA_URL = '{gitlab_url}/api/v4/projects/2596/repository/files/Packs%2F{pack_id}%2Fpack_metadata.json'
+
+
+@lru_cache
+def get_env_var(var_name: str) -> str:
+    """
+    Get an environment variable.
+    This method adds a cache layer to the 'os.getenv' method, and raises an error if the variable is not set.
 
-def is_pack_deprecated(pack_path: str) -> bool:
-    """Checks whether the pack is deprecated.
-    Tests are not being collected for deprecated packs and the pack is not installed in the build process.
     Args:
-        pack_path (str): The pack path
+        var_name (str): Name of the environment variable to get.
+
     Returns:
-        True if the pack is deprecated, False otherwise
+        str: Value of the environment variable.
     """
-    pack_metadata_path = Path(pack_path) / PACK_METADATA_FILENAME
-    if not pack_metadata_path.is_file():
-        return True
-    return tools.get_pack_metadata(str(pack_metadata_path)).get('hidden', False)
+    var_value = os.getenv(var_name)
+    if not var_value:
+        raise ValueError(f"Environment variable '{var_name}' is not set.")
 
+    return var_value
 
-def get_pack_id_from_error_with_gcp_path(error: str) -> str:
+
+@lru_cache(maxsize=128)
+def fetch_pack_metadata_from_gitlab(pack_id: str, commit_hash: str) -> dict:
     """
-        Gets the id of the pack from the pack's path in GCP that is mentioned in the error msg.
+    Fetch pack metadata from master (a commit hash of the master branch when the build was triggered) using GitLab's API.
+
     Args:
-        error: path of pack in GCP.
+        pack_id (str): ID of the pack to fetch metadata for (name of Pack's folder).
+        commit_hash (str): A commit hash to fetch the metadata file from.
 
     Returns:
-        The id of given pack.
+        dict: A dictionary containing pack's metadata.
     """
-    return error.split('/packs/')[1].split('.zip')[0].split('/')[0]
+    api_url = GITLAB_PACK_METADATA_URL.format(gitlab_url=get_env_var('CI_SERVER_URL'), pack_id=pack_id)
+    logging.debug(f"Fetching 'pack_metadata.json' file from GitLab for pack '{pack_id}'...")
+    response = GITLAB_SESSION.get(api_url,
+                                  headers={'PRIVATE-TOKEN': get_env_var('GITLAB_API_READ_TOKEN')},
+                                  params={'ref': commit_hash})
+
+    if response.status_code != 200:
+        logging.error(f"Failed to fetch pack metadata from GitLab for pack '{pack_id}'.\n"
+                      f"Response code: {response.status_code}\nResponse body: {response.text}")
+        response.raise_for_status()
+
+    file_data_b64 = response.json()['content']
+    file_data = base64.b64decode(file_data_b64).decode('utf-8')
 
+    return json.loads(file_data)
 
-def get_pack_display_name(pack_id: str) -> str:
+
+def is_pack_deprecated(pack_id: str, check_locally: bool = True,
+                       commit_hash: str | None = None, pack_api_data: dict | None = None) -> bool:
```

Comment:
Please change the function, move all logic related to the deprecated check inside this function, using my other comments, and add the explanations for the docstrings here.